### PR TITLE
WT-6750 Add an assert to check if the key already exists in the history store

### PR DESCRIPTION
**Pull request by hbokhari**

### CODE REVIEW

File: src/history/hs_rec.c [hunk 1]
```diff
@@ -233,6 +233,21 @@ __hs_insert_record_with_btree(WT_SESSION_IMPL *session, WT_CURSOR *cursor, WT_BT
     start_time_point.durable_ts = upd->durable_ts;
     start_time_point.txnid = upd->txnid;
 
+#ifdef HAVE_DIAGNOSTIC
+    /*
+     * We may have fixed out of order keys. Make sure that we haven't accidentally added a duplicate
+     * of the key we are about to insert.
+     */
+    WT_ERR_NOTFOUND_OK(
+      __wt_hs_cursor_position(session, cursor, btree->id, key, upd->start_ts, srch_key), true);
```
Comment (quchenhao) [inline src/history/hs_rec.c:242]: Can we avoid doing the search here?

Comment (hbokhari) [reply src/history/hs_rec.c:242]: I am not sure what state the cursor would be in after the call to `__hs_fixup_out_of_order_from_pos`. If we preserve the cursor then we can avoid searching again.

Comment (quchenhao) [reply src/history/hs_rec.c:242]: I believe we should be at the cursor position that we will insert the new key as __hs_insert_record_with_btree_int doesn't do any search.

**Comment by hbokhari** [comment]
@quchenhao @bvpvamsikrishna - This is ready for review.

File: src/history/hs_rec.c (hunk 1 above)
Comment (hbokhari) [reply src/history/hs_rec.c]: Fixed.

**Review by quchenhao — APPROVED** [review]
lgtm

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Pull request merged**